### DISCUSSION TITLE
Request: Ability to set `refetchInactive` Default via `queryCache` Configuration

### DISCUSSION THREAD

**Discussion by joealden** [TanStack/query#708] [General] [answered]
First of all, thanks for spending the time to look at this request (and thanks for the awesome library)!

My team is at a point where we are ready to upgrade from v1 to v2 in a project, but I've hit a roadblock with the `refetchQueries` -> `invalidateQueries` changes introduced in v2, specifically the change where now only active queries get refetched. We'd like to be able to keep the behaviour of `refetchQueries` in v1 where all queries (active or not) are refetched. From looking at the v2 docs, I can see that `invalidateQueries` has a `refetchInactive` configuration option, which to my understanding would make `invalidateQueries` act the same as (or similar enough to) `refetchQueries` in v1 when `refetchInactive` is set to `true`.

If this is the case, then this is great, but the problem is that we'd have to set `refetchInactive` to `true` on every call to `invalidateQueries`, which would be less than ideal and prone to mistakes (of people forgetting to set it). For this reason, we'd like to be able to set `refetchInactive` to `true` globally somehow (or I guess at the `queryCache` level) so that the "default" could be `true` instead of `false`.

Would this be something that you'd be happy to support? If so, do you think it would be that complex (or even make sense) to add?

**Comment by tannerlinsley** [comment, marked as answer]
This would be relatively easy to add to the global config.

**Reply by joealden** [reply]
Sorry for the lack of a response on this! FYI, we've decided that for now, due to the flexibility of `react-query`, it likely makes sense for us to solve this in user-land (to not increase the complexity of `react-query` itself where it might not help many people) by creating a wrapper function that provides `refetchInactive: true` automatically where needed.

Thanks for your response though!
